14/README.md

```diff
@@ -0,0 +1,173 @@
+TLS SHALL allow Logging Targets, that is, destinations, for messages.  A Device Server SHALL emit messages to all targets (comma-separated) to its logging_target Property.
+#### console
+output SHALL be delivered to stdout
+#### file
+output SHALL be written to a standard location in a directory specified by TANGO_LOG_PATH if nonblank, otherwise in a directory created under /tmp/tango-*.
+#### file::filename
+output SHALL be written to the filename given.  File format SHALL be log4j compatible.
+#### device::devicename
+output SHALL be sent to the Tango Device given by the device name given.  The 
+The Device running as devicename SHALL implement the Log Consumer interface (see below).
+
+### Logging Levels
+TLS SHALL support Logging Levels.  The Logging Level is meant to act as a filter for the amount of inform.  These are the logging levels, with typical meanings for when they are to be used.
+- OFF: Nothing is logged
+- FATAL: A fatal error occurred. The process is about to abort
+- ERROR: An (unrecoverable) error occurred but the process is still alive
+- WARN: An error occurred but could be recovered locally
+- INFO: Provides information on important actions performed
+- DEBUG: Generates detailed information describing the internal behavior of a device
+
+Levels SHALL be ordered the following way:  
+```DEBUG < INFO < WARN < ERROR < FATAL < OFF```  
+For a given device, a level SHALL be said to be enabled if it is greater or equal to the logging level assigned to this device.
+In other words, any logging request which level is lower than the device's logging level is ignored.
+
+The all the Device's Logging Targets SHALL share the same device logging level.
+
+### C++ API
+TLS SHALL provide the following C++ macros for generating messages.
+
+There SHALL be two forms, printf-like and stream-like.
+
+printf-like: ```LOG_DEBUG(("Msg#%d - Hello world", i++));```
+
+stream-like: ```DEBUG_STREAM << "Msg#" << i++ << "- Hello world" << endl;```
+
+These two logging requests are equivalent. Note the double parenthesis in the printf version.
+
+The examples just given are for the DEBUG log level.  There SHALL BE equivalent macros for the other log levels, except OFF.
+
+TLS SHALL provide a means to log in the name of a device.  This SHALL be provided when the Device must is a member of a class inheriting from the ```Tango::LogAdapter``` class.
+
+### Python API
+TLS SHALL provide the folliwng Python forms of writing log messages:
+
+basic logging: ```self.debug_stream("read_voltage(%s, %d)", self.host, self.port)```
+
+logging with print: ```print >>self.debug_info, "read voltage attribute"```
+
+logging with decorators: ```@PyTango.DebugIt()```
+
+The examples just given are for the DEBUG log level.  There SHALL BE equivalent macros for the other log levels, except OFF.
+
+
+### Java API
+TLS SHALL provide configuration to allow Java to log messages in the following manner:   
+```get_logger().debug(Initializing device  + get_name());```   
```

Comment:
Well, as Java developer I would say 99,9999% of the applications use slf4j in which logger initialization and the command above looks different. So let's just keep this